### ISSUE TITLE
error 'arguments' is not defined `no-undef`

### ISSUE THREAD

**Issue by btnwtn** [eslint/eslint#6853]
**What version of ESLint are you using?**
v3.1.1

**What parser (default, Babel-ESLint, etc.) are you using?**
default/eslint-loader?

**Please show your full configuration:**
```js
{
  "parserOptions": {
    "ecmaVersion": 6
  },
  "env": {
    "browser": true
  },
  "extends": "eslint:recommended",
  "rules": {
    "indent": ["error", 2],
    "quotes": ["error", "single"],
    "semi": ["error", "never"],
    "no-console": "off"
  }
}
```

**What did you do? Please include the actual source code causing the issue.**
Setup a barebones eslint preLoader in webpack, set ecmaVersion to 6 and tried to spread arguments into console.log

```js
const lol = () => console.log(...arguments)
lol([1, 2, 3, 4, 'a', 'b', 'c'])
```

**What did you expect to happen?**
Was expecting no error and for the code to log:

```
# 1, 2, 3, 4, a, b, c
```

**What actually happened? Please include the actual, raw output from ESLint.**
```sh
λ ~/Webpack/t2/ webpack
Hash: 721a56295cd47f43ba06
Version: webpack 1.13.1
Time: 732ms
        Asset    Size  Chunks             Chunk Names
namedEntry.js  1.6 kB       0  [emitted]  namedEntry
    + 1 hidden modules

ERROR in ./src/index.js

/Users/bnewton/Webpack/t2/src/index.js
  3:34  error  'arguments' is not defined  no-undef

✖ 1 problem (1 error, 0 warnings)
```

**Comment by gyandeeps** [comment]
Thats a syntax error when it comes to arrow functions.

> Arrow functions do not expose an arguments object to their code: arguments.length, arguments[0], arguments[1], and so forth do not refer to the arguments provided to the arrow function when called.  

Also if you try to run your code in chrome console directly, it will throw an error.

**Comment by ilyavolodin** [comment]
To the best of my knowledge, you can't use arguments inside arrow functions. It looks like the rule is doing the right thing.

**Comment by btnwtn** [comment]
Ah it appears you two are correct, looks like I need to brush up a bit more on arrow functions. Closing this issue.
